api/src/iot/management/commands/import_csv.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import unicode_literals
+
+import csv
+from pathlib import Path
+
+from django.core.management import BaseCommand
+from django.db import transaction
+from django.db.models import Q
+
+from iot.import_utils import CsvRow, CsvRowValidationException, import_row
+from iot.mail import send_csv_import_report
+from iot.models import Device, Person, Type
+from iot.settings import settings
+
+
+class Command(BaseCommand):
+    """
+    Import a CSV File into the database
+    """
+    help = 'Import a CSV file into the database, this will replace the currently loaded data'
+
+    def add_arguments(self, parser):
+        parser.add_argument('file_path', type=str)
+
+    def handle(self, *args, **options):
+        with transaction.atomic():
+            if self._pre_handle(*args, **options):
+                self._handle(*args, **options)
+            self._post_handle(*args, **options)
+
+    def _pre_handle(self, *args, **options):
+        self.line_count = 0
+        self.success = 0
+        self._errors = []
+        self.csv_file = Path(options['file_path'])
+        self.validation_failed_for = []
+
+        self.stdout.write('CSV File: {}'.format(self.csv_file))
+
+        if not self.csv_file.exists() or not self.csv_file.is_file():
+            self.csv_file = None
+            return False
+
+        with open(str(self.csv_file)) as csv_file:
+            reader = csv.reader(csv_file)
+            self.line_count = sum(1 for _ in reader) - 1
+
+        return True
+
+    def _handle(self, *args, **options):
+        Device.objects.all().delete()
+
```

Comment:
This implies the full dataset is rebuilt each import, right?

Reply:
Yes